app/controllers/orders_controller.rb

```diff
@@ -1,13 +1,24 @@
 class OrdersController < ApplicationController
+  before_action :authenticate_user!
+
   def add_to_order
-    if Order.last
-      order = Order.last
-    else
-      order = Order.create
+	  if session[:order_id] == nil || Order.find(session[:order_id]).finalized == true
+		  create_order
+	  else
+      @order = Order.find(session[:order_id])
     end
+
     dish = Dish.find(params[:dish_id])
-    order.add(dish, dish.price)
+    @order.add(dish, dish.price)
     flash[:notice] = "Successfully added to order"
     redirect_back(fallback_location: root_path)
   end
+
+  private
+
+  def create_order
+    order = Order.create(user: current_user)
```

Comment:
If you do 
```
@order = Order.create(user: current_user)
session[:order_id] = @order.id
``` 
Then you don't need the last line and avoid making another db call.